### PR TITLE
feat: contrax-finance tvl adapter

### PR DESCRIPTION
**Pull request by abdulrafay-contrax**
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Please enable "Allow edits by maintainers" while putting up the PR.
3. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Github org/user (Optional, if your code is open source, we can track activity): https://github.com/Contrax-co

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/contrax-finance exports TVL: 
        
 
 
    arbitrum                  8.44 k
    
    total                    8.44 k

File: projects/contrax-finance/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+async function tvl(_, _1, _2, { api }) {
+  let promises = [];
+  promises = Vaults.map(async (vaultAddr) => {
+    const tokenAddress = await api.call({
+      abi: "function token() view returns (address token)",
+      target: vaultAddr,
+      params: [],
+    });
+    const controllerAddress = await api.call({
+      abi: "function controller() view returns (address controller)",
```
[the remark this answers is not in the document]
Comment (g1nt0ki) [reply projects/contrax-finance/index.js:53]: or can be changed to:


    async function tvl(_, _1, _2, { api }) {
      const tokens = await api.multiCall({  abi: 'address:token', calls: Vaults})
      const bals = await api.multiCall({  abi: 'uint256:balance', calls: Vaults})
      api.addTokens(tokens, bals)
      return sumTokens2({ api, resolveLP: true,})
    }

[the remark this answers is not in the document]
Comment (abdulrafay-contrax) [reply projects/contrax-finance/index.js:53]: Vaults does not store all the tokens, there is treasury and compounding going on. I am able to get correct amount from the controller.
I'll add multicall and clean the code.

**Comment by llamatester** [comment]
The adapter at projects/contrax-finance exports TVL: 
        
 
 
    arbitrum                  8.35 k
    
    total                    8.35 k

**Pull request merged**